### PR TITLE
Enable displaying PHP errors to stderr if running behat.

### LINKED ISSUE #4504: Consider enabling PHP errors by default in Behat tests
From https://github.com/wp-cli/extension-command/pull/62#pullrequestreview-75899079:

> Have a PR to expose PHP notices/warnings when running behat which I'll post one day rather than having to do ad-hoc debug file requires and tags etc...

One caveat when doing so: we may expose a good number of issues we're not currently aware of. Before we commit to this, it'd be good to run a complete test suite with this enabled to see what sort of existing problems we have.

### PR DESCRIPTION
**Pull request by gitlost**
Fixes #4504

Enables displaying PHP errors (except `E_DEPRECATED`) to stderr if running behat, first on starting WP-CLI and then after WP loads.

Changes `WP_CLI\Process` to throw an error if `stderr` is non-empty if running behat.

Changes `FeatureContext` to redirect composer messages (which are output to stderr) to stdout, and to redirect warning messages on caching wp core download, and to `skip-email`, so as not to pollute stderr.

Suppresses "Warning: Wildcard DNS may not be configured correctly" message on `Given a WP multisite install`, so as not to pollute stderr. Also adds `not` option to the `Then the return code should` step for situations where there could be OS-dependent return values.

Adjusts various tests to allow for these changes, and removes individual stderr handling. Also in `help.feature` simplifies some copy-pasta introduced (by me) in https://github.com/wp-cli/wp-cli/pull/4285.

As noted in https://github.com/wp-cli/wp-cli/issues/4504, this PR will cause tests for bundled commands to fail until they're adjusted. It will also cause breakage in any external code that uses the behat test suite.

### CODE REVIEW

**Comment by schlessera** [comment]
This would now be a use case for an external test suite package, that could be separately versioned. This way, we could add breaking change by bumping the major versions. Different commands could use different versions of the tests.

(... and probably break our automated full test suite in the process...? )

**Comment by danielbachhuber** [comment]
> As noted in #4504, this PR will cause tests for bundled commands to fail until they're adjusted.

Is this actually the case though? These files are copied into the bundled commands.

File: php/WP_CLI/Process.php [hunk 1]
```diff
@@ -112,6 +112,10 @@ public function run_check() {
 		if ( $r->return_code || ! empty( $r->STDERR ) ) {
 			throw new \RuntimeException( $r );
 		}
+		// But do it right if testing.
+		if ( ! empty( $this->env['BEHAT_RUN'] ) && ! empty( $r->stderr ) ) {
```
Comment (danielbachhuber) [inline php/WP_CLI/Process.php]: Can we perform this check where `run_check()` is called, instead of modifying the `Process` class itself?

**Comment by gitlost** [comment]
> Is this actually the case though?

Those tests that do an `And I run` step that produce non-empty stderr will fail until adjusted - a simple example is those that use the `--debug` flag, eg in [package-command/features/package-install.feature#L31](https://github.com/wp-cli/package-command/blob/v1.0.8/features/package-install.feature#L31).

File: php/WP_CLI/Process.php (hunk 1 above)
Comment (gitlost) [reply php/WP_CLI/Process.php]: It's called in a lot of places - do mean just in `invoke_proc()` https://github.com/wp-cli/wp-cli/blob/v1.4.1/features/steps/when.php#L7?

**Comment by danielbachhuber** [comment]
> Those tests that do an And I run step that produce non-empty stderr will fail until adjusted

But they're only fail after I run `wp scaffold package-tests` ? e.g. the fail won't occur until I copy the new version of the test suite into the package.

File: php/WP_CLI/Process.php (hunk 1 above)
Comment (danielbachhuber) [reply php/WP_CLI/Process.php]: Yes. Given this change is specific to the test suite, it makes more sense to include it in the test suite.

**Comment by gitlost** [comment]
> But they're only fail after I run wp scaffold package-tests

Sorry, yes (I think!).

File: php/WP_CLI/Process.php (hunk 1 above)
Comment (gitlost) [reply php/WP_CLI/Process.php]: Hmm, that would make the code a bit yuck, which usually isn't a good sign. Is the `BEHAT_RUN` check not exclusive enough?

Comment (danielbachhuber) [reply php/WP_CLI/Process.php]: > Hmm, that would make the code a bit yuck, which usually isn't a good sign.

Why?

> Is the `BEHAT_RUN` check not exclusive enough?

It's hacky. It's not the Process class' responsibility to adapt behavior based on some environmental flag. It's Behat that wants to change the behavior of `STDERR`.

Comment (gitlost) [reply php/WP_CLI/Process.php]: I'll add it as a new function `run_check_stderr()` - that should keep us both happy...

**Comment by gitlost** [comment]
Cool. How do you want to deal with fixing the test breakages in the bundled commands? If you want to split them out and assign them I'm good for several hours now...

**Comment by danielbachhuber** [comment]
> If you want to split them out and assign them I'm good for several hours now...

If you want to go through each package individually and run `wp scaffold package-tests ./ --force`, I'm fine with that. I was going to refrain until #3924 is ready but I'm not opposed to doing them manually.

I don't think you need to create issues though. Feel free to simply start creating PRs with the updated test suite and the fixes.

**Comment by gitlost** [comment]
Okay I'll start going through them alphabetically...

**Pull request merged**